### DISCUSSION TITLE
Drift detected: Your database schema is not in sync with your migration history.

### DISCUSSION THREAD

**Discussion by iamvanja** [prisma/prisma#20939] [Q&A] [answered]
### Bug description

Added 

```sql
model User {
  ...
  role      Role     @default(USER)
  ....
}


enum Role {
  USER
  ADMIN
}
```

Did `prisma db push` (no problem)

When it came to `prisma migrate dev`, I get the following message

```
Drift detected: Your database schema is not in sync with your migration history.

The following is a summary of the differences between the expected database schema given your migrations files, and the actual schema of the database.

It should be understood as the set of changes to get from the expected schema to the actual schema.

[+] Added enums
  - Role

[*] Changed the `User` table
  [+] Added column `role`

✔ We need to reset the "public" schema at "localhost:5432"
Do you want to continue? All data will be lost. … no
```

`prisma migrate dev --create-only` renders the same result (and asks for resetting the db also?)

This happened several times already as I had the big changes. This time I started all over again with a brand new DB. Initial migration worked fine, but then the same issue again. 

### How to reproduce

See above.

### Expected behavior

Database should not be reset for this change.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}

model User {
  id        Int      @id @default(autoincrement())
  email     String   @unique() @db.VarChar(320)
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  role      Role     @default(USER)
}

enum Role {
  USER
  ADMIN
}

```

### Environment & setup

- OS: macOS, Vercel 
- Database: PostgreSQL (locally), Supabase elsewhere
- Node.js version: v18.16.1

### Prisma Version

```
prisma                  : 5.2.0
@prisma/client          : 5.2.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 2804dc98259d2ea960602aca6b8e7fdc03c1758f (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 2804dc98259d2ea960602aca6b8e7fdc03c1758f (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.2.0-25.2804dc98259d2ea960602aca6b8e7fdc03c1758f
Default Engines Hash    : 2804dc98259d2ea960602aca6b8e7fdc03c1758f
Studio                  : 0.494.0
```

**Comment by janpio** [comment, marked as answer]
`db push` and `migrate dev` are two separate systems. The moment you did `db push`, this migration is not logged into a migration file - and `migrate dev` on its next run will try to reconstruct the history by running all of these, and then warns you that the actual database and your history do not match.

Always use `migrate dev` to create migrations and change your database.
If you need to use `db push` for some reason, you will need to use baselining to tell `migrate dev` that this change happened outside of its system.

**Comment by iamvanja** [comment]
Thanks for the quick response @janpio! 🌮💪
Your explanation makes perfect sense.

I must have misunderstood the dev flow . Will need to revisit the docs. I was under the impression that `db push` is part of the db scheming until fully satisfied while developing. Then run `migrate dev` to catch up the migration up to speed. 

But the dev flow instead seems to be either:
- `db push`, use baselining, `migrate dev` or
- always use `migrate dev`, even if means resulting with multiple different migrations as part of that single feature

Did I get this right?

**Reply by janpio** [reply]
Yes. You can use `db push` for quick prototyping, but at the end you need to somehow reconcile for `migrate` -and that works best with baselining.
